src/register/data/utils.js

```diff
@@ -43,39 +43,30 @@ export const isFormValid = (
   Object.keys(payload).forEach(key => {
     switch (key) {
     case 'name':
-      if (!fieldErrors.name) {
-        fieldErrors.name = validateName(payload.name, formatMessage);
-      }
+      fieldErrors.name = validateName(payload.name, formatMessage);
       if (fieldErrors.name) { isValid = false; }
       break;
     case 'email': {
-      if (!fieldErrors.email) {
-        const {
-          fieldError, confirmEmailError, suggestion,
-        } = validateEmail(payload.email, configurableFormFields?.confirm_email, formatMessage);
-        if (fieldError) {
-          fieldErrors.email = fieldError;
-          isValid = false;
-        }
-        if (confirmEmailError) {
-          fieldErrors.confirm_email = confirmEmailError;
-          isValid = false;
-        }
-        emailSuggestion = suggestion;
+      const {
+        fieldError, confirmEmailError, suggestion,
+      } = validateEmail(payload.email, configurableFormFields?.confirm_email, formatMessage);
+      if (fieldError) {
+        fieldErrors.email = fieldError;
+        isValid = false;
+      }
```

Comment:
i think the changes here resolves an existing bug. We should make a ticket for this.

Reply:
Which bug?

Reply:
he function does not check if the field already has an error and overrides the error with frontend validations. For example, if email field has an error of already existing email and we pass email field through isFormValid Function on submit, it will override the error with frontend validations (will only check if email is valid) and make submit call. Instead in actual, the call should not be made when we had an existing  email bug.

Reply:
Address!